libmcount/mcount.h

```diff
@@ -74,6 +74,14 @@ enum shmem_buffer_flags {
 	SHMEM_FL_RECORDING	= (1U << 2),
 };
 
+#define SAVE_ERRNO(block)			\
+do {						\
+	int _saved_errno = errno;		\
+	do { block; } while (0);		\
+	errno = _saved_errno;			\
+}						\
+while (0);
+
```

Comment:
Do you consider using this macro in other place?  If not, I think open coding is better.